homeassistant/components/zwave_js/__init__.py

```diff
@@ -131,8 +131,8 @@ def async_on_node_removed(node: ZwaveNode) -> None:
         # grab device in device registry attached to this node
         dev_id = get_device_id(client, node)
         device = dev_reg.async_get_device({dev_id})
-        # note: removal of entity registry is handled by core
-        dev_reg.async_remove_device(device.id)
+        # note: removal of entity registry entry is handled by core
+        dev_reg.async_remove_device(device.id)  # type: ignore
```

Comment:
Why didn't mypy pick up this problem before? We don't change anything about the device return type typing in the helpers that I can see.

Reply:
I think typing of `async_get_registry` must have been impeded by the `@singleton` decorator which is no longer used in this PR. I didn't dig into it though.

Reply:
Yeah, that was my guess too, but I don't see why that should impact the device typing. :shrug:

Reply:
It impacts the device typing because mypy somehow doesn't know the type of `dev_reg`?